lib/galaxy/datatypes/binary.py

```diff
@@ -306,7 +331,7 @@ class DynamicCompressedArchive(CompressedArchive):
     compressed_format: str
     uncompressed_datatype_instance: Data
 
-    def matches_any(self, target_datatypes) -> bool:
+    def matches_any(self, target_datatypes: List[Any]) -> bool:
```

Comment:
I'm not entirely sure what type this is. Some older typing appears to suggest it could be a list of strings (which it is not, but still). It expects the item in the list to have an attribute, `compressed_format` - and that would make it a `FilePrefix`. I couldn't easily trace this, so I decided to leave this as `Any`. Unless there's an obvious solution, this could lead down a garden path.. My vote is to tighten it in another PR.